src/dataframe/dataframe.jl

```diff
@@ -685,8 +692,16 @@ function insertcols!(df::DataFrame, col_ind::Int, name_cols::Pair{Symbol,<:Any}.
 
     for (name, item) in name_cols
         if !(item isa AbstractVector)
-            @assert !(item isa AbstractArray)
-            item_new = fill!(Tables.allocatecolumn(typeof(item), target_row_count), item)
+            if item isa AbstractArray
```

Comment:
How about merging the first two branches and doing `@assert item isa AbstractArray{<:Any, 0}`?

Reply:
Good point. I merged them and made an assert in the following branch. Essentially we want to disallow `AbstractArray` of dimension larger than 1.